drivers/ethernet/phy_xlnx_gem.c

```diff
@@ -0,0 +1,975 @@
+		PHY_MRVL_GENERAL_CONTROL_1_REGISTER, phy_data);
+
+	/* Bit [15] reverts to 0 once the reset is complete. */
+	while ((phy_data & PHY_MRVL_COPPER_CONTROL_RESET_BIT) != 0 &&
+		retries++ < 10) {
+		phy_data = phy_xlnx_gem_mdio_read(dev_conf->base_addr,
+			dev_data->phy_addr,
+			PHY_MRVL_GENERAL_CONTROL_1_REGISTER);
+	}
+	if (retries == 10) {
+		LOG_ERR("%s configure PHY address %hhu (Marvell Alaska) timed out",
+			dev->name, dev_data->phy_addr);
+		return;
+	}
+	retries = 0;
+
+	/* Revert to register page 0 */
+	phy_xlnx_gem_mdio_write(dev_conf->base_addr, dev_data->phy_addr,
+		PHY_MRVL_COPPER_PAGE_SWITCH_REGISTER,
+		PHY_MRVL_BASE_REGISTERS_PAGE);
+
+	/*
+	 * Configure MDIX
+	 * TODO: Make this value configurable via KConfig or DT?
+	 * Page 0, register address 16 = Copper specific control register 1,
+	 * bits [6..5] = MDIO crossover mode. Comp. datasheet table 76.
+	 * NOTICE: a change of this value requires a subsequent software
+	 * reset command via Copper Control Register's bit [15].
+	 */
+
+	/* [6..5] 11 = Enable auto cross over detection */
+	phy_data  = phy_xlnx_gem_mdio_read(dev_conf->base_addr,
+		dev_data->phy_addr, PHY_MRVL_COPPER_CONTROL_1_REGISTER);
+	phy_data &= ~(PHY_MRVL_MDIX_CONFIG_MASK << PHY_MRVL_MDIX_CONFIG_SHIFT);
+	phy_data |= (PHY_MRVL_MDIX_AUTO_CROSSOVER_ENABLE <<
+		PHY_MRVL_MDIX_CONFIG_SHIFT);
+	phy_xlnx_gem_mdio_write(dev_conf->base_addr, dev_data->phy_addr,
+		PHY_MRVL_COPPER_CONTROL_1_REGISTER, phy_data);
+
+	/*
+	 * Configure the Copper Interrupt Enable Register.
+	 * The interrupt status register provides a convenient way to
+	 * detect relevant state changes, also, PHY management could
+	 * eventually be changed from polling to interrupt-driven.
+	 * There's just one big catch: at least on the Zedboard, the
+	 * PHY interrupt line isn't wired up, therefore, the GEM can
+	 * never trigger a PHY interrupt. Still, the PHY interrupts
+	 * are configured & enabled in order to obtain all relevant
+	 * status data from a single source.
+	 *
+	 * -> all bits contained herein will be retained during the
+	 * upcoming software reset operation.
+	 * Page 0, register address 18 = Copper Specific Interrupt
+	 * Enable Register,
+	 * bit [14] = Speed changed interrupt enable,
+	 * bit [13] = Duplex changed interrupt enable,
+	 * bit [11] = Auto-negotiation completed interrupt enable,
+	 * bit [10] = Link status changed interrupt enable.
+	 * Comp. datasheet table 78
+	 */
+	phy_data = PHY_MRVL_COPPER_SPEED_CHANGED_INT_BIT |
+		PHY_MRVL_COPPER_DUPLEX_CHANGED_INT_BIT |
+		PHY_MRVL_COPPER_AUTONEG_COMPLETED_INT_BIT |
+		PHY_MRVL_COPPER_LINK_STATUS_CHANGED_INT_BIT;
+	phy_xlnx_gem_mdio_write(dev_conf->base_addr, dev_data->phy_addr,
+		PHY_MRVL_COPPER_INT_ENABLE_REGISTER, phy_data);
+
+	/*
+	 * Page 0 / Reg 0 [15] Copper Software Reset bit, affecting pages
+	 * 0, 2, 3, 5, 7. Reset is performed immediately, bit [15] is
+	 * self-clearing.
+	 */
+	phy_data  = phy_xlnx_gem_mdio_read(dev_conf->base_addr,
```

Comment:
can this be reworked so that you can use the alaska_reset() function above instead of open coding this?